### PR TITLE
reenable collection reflection tests, with impl details ignored

### PR DESCRIPTION
**Pull request by Gankra**
We still want to be able to verify the reflection system doesn't crash on these types, so let's just make sure the high level details hold.

Note that this is still relying on implementation details of these types which are slated to change. Each of these types has extra size and lacks extra inhabitants due to the use of an enum at their root. The eager bridging PRs change this. That said, tracking size/align/inhabitants isn't especially burdensome, and is probably worth doing. Although it's easy for these details to slip past CI since there's 32-bit and 64-bit configurations.

On that note: any easy way to have the CI check this PR on 32 bit? Looks like some memory layout details changed since these were turned off!

CC @dabrahams and @slavapestov

### CODE REVIEW

**Comment by Gankro** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@shahmishal might be able to answer the 32 bit CI question.

**Comment by Gankro** [comment]
looks like full test does 32-bit!

**Comment by Gankro** [comment]
@swift-ci Please test

**Comment by Gankro** [comment]
@swift-ci Please test

**Comment by Gankro** [comment]
@swift-ci Please test

**Comment by Gankro** [comment]
@swift-ci Please test

File: validation-test/Reflection/reflect_Array.swift [hunk 1]
```diff
@@ -26,15 +23,10 @@ reflect(object: obj)
...
+// CHECK-64: (class_instance size=24 alignment=8 stride=24 num_extra_inhabitants=0
 // CHECK-64:   (field name=t offset=16
 // CHECK-64:     (struct size=8 alignment=8 stride=8 num_extra_inhabitants=1
-// CHECK-64:       (field name=_buffer offset=0
-// CHECK-64:         (struct size=8 alignment=8 stride=8 num_extra_inhabitants=1
-// CHECK-64:           (field name=_storage offset=0
-// CHECK-64:             (struct size=8 alignment=8 stride=8 num_extra_inhabitants=1
-// CHECK-64:               (field name=rawValue offset=0
-// CHECK-64:                 (builtin size=8 alignment=8 stride=8 num_extra_inhabitants=1)))))))))
+// (unstable implementation details omitted)
```
Comment (slavapestov) [inline validation-test/Reflection/reflect_Array.swift]: Thanks, this is a good compromise.

**Review by slavapestov — APPROVED** [review]

**Comment by Gankro** [comment]
@dabrahams are you ok with committing to this level of implementation detail? This biggest fallout is my eager bridging PRs, but your string work will probably break these once or twice.

**Comment by dabrahams** [comment]
@Gankro in the interest of progress, I'm going to say "yes" for now.  You've made things better.  If I end up breaking more of these tests I'll rip out more of the detail.

**Pull request merged**